protocol/contracts/nexus/Shards.sol

```diff
@@ -0,0 +1,232 @@
+                transmutation.requiredAmounts
+        );
+        // Create
+        
+        _mintBatch(
+                msg.sender,
+                transmutation.rewardIds,
+                transmutation.rewardAmounts,
+                ""
+            );      
+        emit Transmutation(msg.sender, _recipeId);
+    }
+
+     // For partners to mint their items, logic implemented on their side
+    function partnerMint(
+        address _to,
+        uint256 _id,
+        uint256 _amount,
+        bytes memory data
+    ) external {
+        require(partnerMinters[msg.sender], "You're not authorised to mint");
+        require(partnerAllowedIds[msg.sender][_id], "This isn't your reserved itemId");
+        _mint(_to, _id, _amount, data);
+        emit MintedShard(_to, _id, _amount);
+    }
+
+    function uri(uint256 _id) override public view returns(string memory){
+        return tokenURIs[_id];
+    }
+
+    //------- Owner -------//
+
+    // authorise a partner to mint 
+    function addPartner(address _partnerMinter, bool _flag) external onlyOwner {
+        partnerMinters[_partnerMinter]=_flag;
+    }
+    
+    // authorise specific shard ids to be minted by a partner
+    function whiteListItemsForPartner(address _partner, uint256[] memory _allowedIds, bool _allowed) external onlyOwner{
+        for(uint i = 0;i<_allowedIds.length;i++){
+            partnerAllowedIds[_partner][_allowedIds[i]]=_allowed;
+        }
+    }
+
+    function setRelic(address _relic) external onlyOwner {
+        RELIC = IRelic(_relic);
+    }
+
+    function createRecipe(
+        uint256 _recipeId,
+        uint256[] memory _requiredIds,
+        uint256[] memory _requiredAmounts,
+        uint256[] memory _rewardIds,
+        uint256[] memory _rewardAmounts
+    ) external onlyOwner {
+        recipes[_recipeId].id = uint16(_recipeId);
+        recipes[_recipeId].requiredIds = _requiredIds;
+        recipes[_recipeId].requiredAmounts = _requiredAmounts;
+        recipes[_recipeId].rewardIds = _rewardIds;
+        recipes[_recipeId].rewardAmounts = _rewardAmounts;
+    }
+
+    function mint(
+        address account,
+        uint256 id,
+        uint256 amount,
+        bytes memory data
+    ) public onlyOwner {
+        _mint(account, id, amount, data);
+    }
+
+    function mintBatch(
+        address to,
+        uint256[] memory ids,
+        uint256[] memory amounts,
+        bytes memory data
+    ) public onlyOwner {
+        _mintBatch(to, ids, amounts, data);
+    }
+
+    function _beforeTokenTransfer(
+        address operator,
+        address from,
+        address to,
+        uint256[] memory ids,
+        uint256[] memory amounts,
+        bytes memory data
+    ) internal override(ERC1155, ERC1155Supply) whenNotPaused {
+        super._beforeTokenTransfer(operator, from, to, ids, amounts, data);
+    }
+
+    function setURI(string memory _newUri, uint256 _index) public onlyOwner {
+        tokenURIs[_index] = _newUri;
```

Comment:
So each shard needs to have this called and set independently? After it's minted? What's the flow for calling this?